### PR TITLE
Don't capture 50x errors

### PR DESCRIPTION
**Pull request by cweagans**
## The Problem:

Capturing 50x errors in the nginx config hides the actual problem that caused the 500 error. Instead, we should just let the user see the original error page so that they can debug appropriately.

## The Fix:

Removed the nginx config that captures 50x output

## The Test:

Run `make test` and verify that everything passes. You can also run the container manually with `-e "DOCROOT=docroot"` and go to localhost/test/500.php to verify that a 500 status code does not get replaced by our messaging.

## Automation Overview:

The automated tests now check a handful of error states for the container.

## Related Issue Link(s):

https://github.com/drud/ddev/issues/537

### CODE REVIEW

**Comment by rfay** [comment]
Please take a look at https://github.com/drud/docker.nginx-php-fpm-local/pull/24 so that we remove the extraneous html files (and anything else we don't want)

**Comment by cweagans** [comment]
@rfay this is ready for another look.

File: test/containertest.sh [hunk 1]
```diff
@@ -19,6 +16,24 @@ echo "starting container for tests"
 docker run -p $HOST_PORT:$CONTAINER_PORT -e "DOCROOT=docroot" -d --name web-local-test -d `awk '{print $1}' .docker_image`
 CONTAINER_NAME=web-local-test ./test/containercheck.sh
 
+function cleanup {
+	echo "Removing web-local-test"
+	docker stop web-local-test 2>/dev/null
```
Comment (rfay) [inline test/containertest.sh]: Simpler to do a docker rm -f instead of the two commands.

File: test/containertest.sh [hunk 2]
```diff
@@ -27,4 +42,4 @@ docker stop web-local-test && docker rm web-local-test
 echo "testing use of custom nginx config"
 docker run -p $HOST_PORT:$CONTAINER_PORT -e "DOCROOT=potato" -v `pwd`/test/test-custom.conf:/var/www/html/.ddev/nginx-site.conf -d --name web-local-test -d `awk '{print $1}' .docker_image`
 docker exec -it web-local-test cat /etc/nginx/sites-enabled/nginx-site.conf | grep "docroot is /var/www/html/potato in custom conf"
-docker stop web-local-test && docker rm web-local-test
+
```
Comment (rfay) [inline test/containertest.sh]: How about a cleanup() at the end?

**Comment by rfay** [comment]
I tested this with ddev and drupal manually by 
1. breaking the db credentials in settings.local.php for a 503
2. Trying to access admin/ when not logged in for a 403
3. Hitting a nonexistent test.php for a 404. Drupal's so slap-happy about redirecting to main for everything it doesn't find that I didn't get a particularly good test.

File: test/containertest.sh (hunk 2 above)
Comment (cweagans) [reply test/containertest.sh]: `cleanup` is called automatically when the script exits at any point via the `trap` statement just below the function body. This ensures that the cleanup function is run at the end if everything works properly, and right after the last non-zero exit code if something is wrong.

File: test/containertest.sh (hunk 1 above)
Comment (cweagans) [reply test/containertest.sh]: Fixed in b2b056f

File: test/containertest.sh (hunk 2 above)
Comment (rfay) [reply test/containertest.sh]: Good to go then, thanks!

**Pull request merged**